sentry/src/main/java/io/sentry/SentryMeasurementUnit.java

```diff
@@ -0,0 +1,109 @@
+package io.sentry;
+
+import java.util.Locale;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * <a href="https://getsentry.github.io/relay/relay_metrics/enum.MetricUnit.html#">Develop Docs</a>
+ */
+public interface SentryMeasurementUnit {
+
+  enum Duration implements SentryMeasurementUnit {
+    /** Nanosecond (`"nanosecond"`), 10^-9 seconds. */
+    NANOSECOND,
+
+    /** Microsecond (`"microsecond"`), 10^-6 seconds. */
+    MICROSECOND,
+
+    /** Millisecond (`"millisecond"`), 10^-3 seconds. */
+    MILLISECOND,
+
+    /** Full second (`"second"`). */
+    SECOND,
+
+    /** Minute (`"minute"`), 60 seconds. */
+    MINUTE,
+
+    /** Hour (`"hour"`), 3600 seconds. */
+    HOUR,
+
+    /** Day (`"day"`), 86,400 seconds. */
+    DAY,
+
+    /** Week (`"week"`), 604,800 seconds. */
+    WEEK;
+  }
+
+  enum Information implements SentryMeasurementUnit {
+    /** Bit (`"bit"`), corresponding to 1/8 of a byte. */
+    BIT,
+
+    /** Byte (`"byte"`). */
+    BYTE,
+
+    /** Kilobyte (`"kilobyte"`), 10^3 bytes. */
+    KILOBYTE,
+
+    /** Kibibyte (`"kibibyte"`), 2^10 bytes. */
+    KIBIBYTE,
+
+    /** Megabyte (`"megabyte"`), 10^6 bytes. */
+    MEGABYTE,
+
+    /** Mebibyte (`"mebibyte"`), 2^20 bytes. */
+    MEBIBYTE,
+
+    /** Gigabyte (`"gigabyte"`), 10^9 bytes. */
+    GIGABYTE,
+
+    /** Gibibyte (`"gibibyte"`), 2^30 bytes. */
+    GIBIBYTE,
+
+    /** Terabyte (`"terabyte"`), 10^12 bytes. */
+    TERABYTE,
+
+    /** Tebibyte (`"tebibyte"`), 2^40 bytes. */
+    TEBIBYTE,
+
+    /** Petabyte (`"petabyte"`), 10^15 bytes. */
+    PETABYTE,
+
+    /** Pebibyte (`"pebibyte"`), 2^50 bytes. */
+    PEBIBYTE,
+
+    /** Exabyte (`"exabyte"`), 10^18 bytes. */
+    EXABYTE,
+
+    /** Exbibyte (`"exbibyte"`), 2^60 bytes. */
+    EXBIBYTE;
+  }
+
+  enum Fraction implements SentryMeasurementUnit {
+    /** Floating point fraction of `1`. */
+    RATIO,
+
+    /** Ratio expressed as a fraction of `100`. `100%` equals a ratio of `1.0`. */
+    PERCENT;
+  }
+
+  final class Custom implements SentryMeasurementUnit {
```

Comment:
I'm not entirely sure about this, would require creating a new class instance for every custom unit. If it was Kotlin, I'd have used inline classes, problem solved.

In theory we could provide one more `setMeasurement` overload which accepts a plain String, but also not sure if it's better.

Reply:
I'd keep it as it is, Relay does the same.
Some languages don't have overloads, so at least we keep it unified.

Reply:
what kind of comments would you like to add? There's a link to the develop docs explaining every unit though

Reply:
I've added some comments, mainly deriving from the Relay docs, lemme know if that works.